pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/generator/ColumnPartitionConfig.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.indexsegment.generator;
+
+import com.linkedin.pinot.core.data.partition.PartitionFunction;
+import com.linkedin.pinot.core.data.partition.PartitionFunctionFactory;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.math.IntRange;
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+
+
+@SuppressWarnings("unused") // Suppress incorrect warnings as methods used for ser/de.
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ColumnPartitionConfig {
+  public static final String PARTITION_VALUE_DELIMITER = ",";
+
+  String _functionName;
```

Comment:
It can be if use following C'tor signature. That way you need only one. 
(good to have; not absolutely mandatory)

public ColumnPartitionConfig(@Nonnull @JsonProperty("functionName") String functionName, @JsonProperty("partitionValues") String partitionValues)